### PR TITLE
PNDA-1960 - Make Kafkat available on nodes as option for Kafka management at CLI

### PR DESCRIPTION
**Pull request by boopalans**
PNDA-1960 - Make Kafkat available on nodes as option for Kafka management at CLI

### CODE REVIEW

File: salt/kafka-tool/init.sls [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#read all pillar data
+
+{% set install_dir = pillar['pnda']['homedir'] %}
+{% set packages_server = pillar['packages_server']['base_uri'] %}
+{% set release_directory = salt['pillar.get']('kafkatool:release_dir', '/opt/pnda') %}
+{% set release_version = salt['pillar.get']('kafkatool:version', '0.3.0')  %}
```
Comment (jgarnier) [inline salt/kafka-tool/init.sls]: should be kafkatool:release_version with default value to v0.2.0

File: salt/kafka-tool/init.sls [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      - {{ pillar['g++']['package-name'] }}
+      - {{ pillar['patch']['package-name'] }}
+
+{% if grains['os'] == 'Ubuntu' %}
+kafka-tool-install_build_essential:
+  pkg.installed:
+    - name: build-essential
+{% endif %}
+
+# gem install kafakt-0.3.0
```
Comment (jgarnier) [inline salt/kafka-tool/init.sls]: remove the comment, wrong and not needed

File: salt/kafka-tool/init.sls [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+kafka-tool-install_gem_kafkat:
+  cmd.run:
+    - name: gem install {{ release_directory }}/kafka-tool/kafkat-{{ release_version }}.gem
```
Comment (jgarnier) [inline salt/kafka-tool/init.sls]: the release version of kafka-tool is v0.2.0 but the gem file is kafka-0.2.0.gem and not kafka-v0.2.0.gem

File: salt/kafka-tool/init.sls [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      kafkat_log_path: {{ local_kafkat_log_path }}
+      zk_ip_list: {{ zk_ips }}
+
+kafka-tool-cfg_create_link:
+  file.symlink:
+    - name: /etc/kafkatcfg
+    - target: {{ release_directory }}/kafka-tool/kafkatcfg
+
+kafka-tool-append_path_bashrc:
+  file.append:
```
Comment (jgarnier) [inline salt/kafka-tool/init.sls]: would prefer create a symlink in /usr/bin

File: salt/kafka-tool/init.sls (hunk 1 above)
Comment (boopalans) [reply salt/kafka-tool/init.sls]: Modified

File: salt/kafka-tool/init.sls (hunk 2 above)
Comment (boopalans) [reply salt/kafka-tool/init.sls]: removed

File: salt/kafka-tool/init.sls (hunk 3 above)
Comment (boopalans) [reply salt/kafka-tool/init.sls]: This addressed as part of build.

File: salt/kafka-tool/init.sls (hunk 4 above)
Comment (boopalans) [reply salt/kafka-tool/init.sls]: Created sym link

File: salt/kafka-tool/init.sls [hunk 5]
```diff
@@ -0,0 +1,75 @@
+#read all pillar data
+
+{% set install_dir = pillar['pnda']['homedir'] %}
+{% set packages_server = pillar['packages_server']['base_uri'] %}
+{% set release_directory = salt['pillar.get']('kafkatool:release_dir', '/opt/pnda') %}
+{% set release_version = salt['pillar.get']('kafkatool:version', 'v0.2.0')  %}
```
Comment (jgarnier) [inline salt/kafka-tool/init.sls]: should be release_version

File: salt/kafka-tool/init.sls [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      kafkat_log_path: {{ local_kafkat_log_path }}
+      zk_ip_list: {{ zk_ips }}
+
+kafka-tool-cfg_create_link:
+  file.symlink:
+    - name: /etc/kafkatcfg
+    - target: {{ release_directory }}/kafka-tool/kafkatcfg
+
+#by default kafkat is installed in /usr/local/bin, this is not in $PATH for RHEL. Create sym link in /usr/bin
+{% if grains['os'] == 'RedHat' %}
```
Comment (jgarnier) [inline salt/kafka-tool/init.sls]: why only specific for RedHat, also apply to Ubuntu

Comment (boopalans) [reply salt/kafka-tool/init.sls]: For Ubuntu, /usr/local/bin is already in PATH, and works with out any issue.

File: salt/kafka-tool/init.sls (hunk 5 above)
Comment (boopalans) [reply salt/kafka-tool/init.sls]: Fixed

File: salt/kafka-tool/init.sls [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+{% set install_dir = pillar['pnda']['homedir'] %}
+{% set packages_server = pillar['packages_server']['base_uri'] %}
+{% set release_directory = salt['pillar.get']('kafkatool:release_dir', '/opt/pnda') %}
+{% set release_version = salt['pillar.get']('kafkatool:release_version', 'v0.2.0')  %}
+{% set release_filename = 'kafka-tool-' + release_version + '.tar.gz' %}
+
+{% set p  = salt['pillar.get']('kafka', {}) %}
+{% set local_kafka_path = p.get('prefix', '/opt/pnda/kafka') %}
+{% set local_kafkat_log_path = salt['pillar.get']('kafkatool:log_dir', '/var/log/pnda/kafkatool') %}
```
Comment (jgarnier) [inline salt/kafka-tool/init.sls]: if used on kafkat configuration file, should be kafka log dir in order to clean the indexes (as defined in the kafka pillar, which is /var/kafka-logs)

**Pull request merged**